code/modules/surgery/organs/augments_arms.dm

```diff
@@ -604,3 +604,75 @@
 	if(emp_proof)
 		return
 	muscle_implant.emp_act(severity, owner)
+
+/obj/item/organ/internal/cyberimp/arm/power_cord_cell
+	name = "Cell-compatible power adapter implant"
+	desc = "This is a recently developed implant for installation inside IPCs in order to allow them to easily collect energy from cells"
+	origin_tech = "materials=3;biotech=2;powerstorage=3"
+	contents = newlist(/obj/item/power_cord_cell)
+	requires_robotic_bodypart = TRUE
+
+/obj/item/organ/internal/cyberimp/arm/power_cord_cell/emp_act(severity)
+// Allowing to repair implant
+	if(emp_proof)
+		return
+	damage = 1
+	crit_fail = TRUE
+
+/obj/item/organ/internal/cyberimp/arm/power_cord_cell/surgeryize()
+	if(crit_fail && owner)
+		to_chat(owner, "<span class='notice'>Your [src] feels functional again.</span>")
+	crit_fail = FALSE
+
+/obj/item/power_cord_cell
```

Comment:
Could probably rename that path name so its not the same as the implant one.

Reply:
> Could probably rename that path name so its not the same as the implant one.

dun